### PR TITLE
Add category functionality

### PR DESCRIPTION
**Pull request by ShyamKrishna33**

### CODE REVIEW

File: src/main/java/budgetbuddy/categories/Category.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package budgetbuddy.categories;
+
+public enum Category {
```
Comment (Vavinan) [inline src/main/java/budgetbuddy/categories/Category.java:3]: Good job on using enum class to facilitate category

**Review by Vavinan — CHANGES_REQUESTED** [review]
Overall good job. Consider changing OTHER to OTHERS in enum class Category

File: src/main/java/budgetbuddy/categories/Category.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    DINING(1, "Dining"),
+    GROCERIES(2, "Groceries"),
+    UTILITIES(3, "Utilities"),
+    TRANSPORTATION(4, "Transportation"),
+    HEALTHCARE(5, "Healthcare"),
+    ENTERTAINMENT(6, "Entertainment"),
+    RENT(7, "Rent"),
+    SALARY(8, "Salary"),
+    OTHER(9, "Other");
```
[the remark this answers is not in the document]
Comment (ShyamKrishna33) [reply src/main/java/budgetbuddy/categories/Category.java]: I made the necessary changes.

**Review by Vavinan — APPROVED** [review]
Good job

**Pull request merged**